dnscache/manager.go

```diff
@@ -96,6 +105,21 @@ func (m *DnsCacheManager) doCachedDial(d *net.Dialer, ctx context.Context, netwo
 		return safeDial(address, "")
 	}
 
+	if m.strategy == config.NoCacheStrategy {
+		if len(ips) > 1 {
+			m.cacheStorage.Delete(host)
+			return safeDial(ips[0]+":"+port, "")
+		}
+	}
+
+	if m.strategy == config.RandomStrategy {
+		if len(ips) > 1 {
+			rand.Seed(int64(len(ips)))
```

Comment:
seeding it every time with `len(ips)` would make `ips[rand.Intn(len(ips))]` to return the same IP every time.
also, `rand.Seed` this is something you want to do only once as it is global seed on `rand` package level.

Reply:
yeah, I've fixed it in this commit - 2dae7488979dcd54c6cd38f7fd26037d990905af .
Now rand is lazyly initialized with the first access to randomStrategy handler. Current seed should be enought for the require level of randomness.